src/transformers/trainer_tf.py

```diff
@@ -221,7 +228,9 @@ def create_optimizer_and_scheduler(self, num_training_steps: int):
             self.optimizer, self.lr_scheduler = create_optimizer(
                 self.args.learning_rate,
                 num_training_steps,
-                self.args.warmup_steps,
+                self.args.warmup_steps
+                if self.args.warmup_steps > 0
+                else math.ceil(num_training_steps * self.args.warmup_ratio),
```

Comment:
Small nit: this is not super readable so it would be great to define before this function
```
wamup_steps = self.args.warmup_steps if self.args.warmup_steps > 0 else math.ceil(num_training_steps * self.args.warmup_ratio)
```
(which will be restyled since it's a bit long so you will need to run `make style`) and then use `warmup_steps` here.